Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -261,54 +340,40 @@ def proximal(self, x, tau, out = None):
                 print("Stop at {} iterations with tolerance {} .".format(k, error))
             else:
                 print("Stop at {} iterations.".format(k))                
-            
-        if out is None:                        
-            self.gradient.adjoint(tmp_q, out=tmp_x)
-            tmp_x *= tau
-            tmp_x *= self.regularisation_parameter 
-            x.subtract(tmp_x, out=tmp_x)
-            return self.projection_C(tmp_x)
-        else:          
-            self.gradient.adjoint(tmp_q, out = out)
-            out*=tau
-            out*=self.regularisation_parameter
-            x.subtract(out, out=out)
-            self.projection_C(out, out=out)
+                    
+        self.gradient.adjoint(tmp_q, out=tmp_x)
+        tmp_x *= tmp_tau
+        tmp_x *= self.regularisation_parameter 
```

Comment:
`tmp_tau` and `self.regularisation_parameter` are scalars can they be multiplied together first to reduce computation.

Reply:
There are two cases where `tau` is a `Number` or `DataContainer` (for the preconditioning case):

https://github.com/TomographicImaging/CIL/blob/fcbe5f25d3e6e952a4240a8ed0a492213d678324/Wrappers/Python/cil/optimisation/functions/TotalVariation.py#L309-L313

https://github.com/TomographicImaging/CIL/blob/fcbe5f25d3e6e952a4240a8ed0a492213d678324/Wrappers/Python/cil/optimisation/functions/TotalVariation.py#L358-L362